### ISSUE TITLE
3.1.X and @next emitting declaration file with initializer in ambient context

### ISSUE THREAD

**Issue by southpolesteve** [Microsoft/TypeScript#28230]
**TypeScript Version:**  typescript@3.2.0-dev.20181030

**Search Terms:**
"ambient context", "ambient context initializer", "ambient context public static"

**Code**

```ts
export class Foo {
  public static readonly bar = 0;
  public baz = 0;
}
```

**Expected behavior:**

Typescript emits a definition file without initializers:

```ts
export declare class Foo {
    static readonly bar: number;
    baz: number;
}
```

This is the output created by typescript@3.0.3

**Actual behavior:**

Typescript emits a definition file with initializer:

```ts
export declare class Foo {
    static readonly bar = 0;
    baz: number;
}
```

**Playground Link:**
https://www.typescriptlang.org/play/#src=export%20declare%20class%20Foo%20%7B%0D%0A%20%20%20%20static%20readonly%20bar%20%3D%200%3B%0D%0A%20%20%20%20baz%3A%20number%3B%0D%0A%7D

**Related Issues:**

**Comment by weswigham** [comment]
Actually, we explicitly added support for constant value read-only initializers in ambient contexts to bring them in line with `const` declarations, which already used and supported them.

**Comment by weswigham** [comment]
Any time we add a new typespace construct (like import types) we've had the potential to emit declaration files which could have broken consumers using older compiler versions. You can usually work around the changes and get a compatible .d.ts by annotating the types yourself, thereby preventing the compiler from generating types/initializers.

**Comment by southpolesteve** [comment]
I understand this is working as intended and I do agree with the rationale behind it. 

That said, I still think this is a poor experience for library authors. I would guess that many TS library authors are not baselining their .d.ts. So as an author, how do I discover this issue? Ship code that breaks users? That's what I did. If a foot-gun exists, ideally some warning exists before I've already shot myself in the foot 😄 Some ideas:

- Include info about this issue at https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html Is there a better location I'm not seeing?
- Add potentially breaking .d.ts changes to the release notes
- Tooling to test libs from a consumer perspective against multiple versions of typescript. I get this is a big ask and likely out scope for the main TS repo.

Let me know if you have any thoughts on these suggestions. The first point I may take a crack at since it is just a documentation change.

@DanielRosenwasser API Extractor is something I've actively trying to adopt. Ran into a couple issues: https://github.com/Microsoft/web-build-tools/issues/663 and https://github.com/Microsoft/web-build-tools/issues/922 (Currently blocking me from using api-extractor). The team working on it has been helpful and I'm hopeful we can work through the issues.
